### DISCUSSION TITLE
Issue with despawn_recursive() combined with parent.spawn_scene()

### DISCUSSION THREAD

**Discussion by Indy2222** [bevyengine/bevy#5141] [Q&A]
I am recursively despawning an entity in a system executed during `CoreStage::PreUpdate`:

```rust
commands.entity(entity).despawn_recursive();
```

In a different system, executed during `CoreStage::Update`, I am accessing that entity and spawning an (already loaded) scene as a child. The code below executes (several frames) before the code above:

```rust
commands
    .entity(entity)
    .insert(Ready)
    .with_children(|parent| {
        parent.spawn_scene(cache.get(*object_type).scene());
    });
```

This sometimes leads to a panic: `thread 'main' panicked at 'Entity 1648v0 does not exist'`.

I suspect that this is due to `parent.spawn_scene()` taking long time and somehow accessing the entity.

Could someone point me out to what I am doing wrong or how to solve that situation?

**Comment by DJMcNab** [comment]
I believe this to be a duplicate of https://github.com/bevyengine/bevy/issues/3195

**Reply by Indy2222** [reply]
Thank you for pointing out that issue. It seems very similar in my case. 

However, the scene is already loaded at the time I spawn it (`server.get_load_state(handle)` is `LoadState::Loaded` for the given handle) and the error happens even several seconds after I spawn the scene. It seems very non-deterministic, because most of the time I can do the spawn-despawn in a quick succession with no problem.

**Reply by Indy2222** [reply]
I need to apologies, my previous comment was not entirely correct.

I executed the two systems twice (probably during two consecutive frames) due to the fact that I got twice the same key `keys.get_just_pressed().last()` from `Input<KeyCode>`. Which, by itself, seems like strange behavior.

Thus it seems that my issue is really a manifestation of #3195.
